tools/python/boutiques/localExec.py

```diff
@@ -278,41 +358,39 @@ def execute(self, mount_strings):
                 os.remove(dsname)
         # Check for output files (note: the path-template
         # can contain value-keys)
-        print('Looking for output files:')
+        missing_files = []
+        output_files = []
         for outfile in self.desc_dict['output-files']:
```

Comment:
I think we can use `evaluate` for a lot of this loop, no? I don't remember all of the functionality built in to it, but it is able to at least query outputs for optional vs required, breaking this into a couple smaller loops. If you want I can implement this and merge into your fork.